### ISSUE TITLE
Arg is only Serializable if T is serializable

### ISSUE THREAD

**Issue by uschi2000** [palantir/safe-logging#4]
https://github.com/palantir/safe-logging/blob/develop/safe-logging/src/main/java/com/palantir/logsafe/Arg.java#L23

**Comment by Regan-Koopmans** [comment]
This is true and probably worth addressing, however it would prevent us from creating `SafeArg`s of `CompletableFuture`s, a use-case demonstrated in tests  ([source](https://github.com/palantir/safe-logging/blob/develop/preconditions-assertj/src/test/java/com/palantir/logsafe/testing/LoggableArgRepresentationTest.java#L39)). What do you think @carterkozak ?
